### PR TITLE
updated homework 3

### PR DESCRIPTION
**Pull request by Lubovgroma**

### CODE REVIEW

File: homework3/index.js [hunk 1]
```diff
@@ -1,28 +1,22 @@
 //---------------task 1----------------
 function durationBetweenDates(firstDate, secondDate, dimension) {
```
Comment (sharkich) [inline homework3/index.js]: no default values

File: homework3/index.js [hunk 2]
```diff
@@ -1,28 +1,22 @@
...
-    if (dimension === 'hours') {
-        const diffHours = Math.ceil(diffTime / (1000 * 60));
-        return console.log(diffHours);
-    }
-    if (dimension === 'days') {
-        const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 *24));
-        return console.log(diffDays);
-    }
-    //console.log(duration);
+    return console.log('Wrong dimension');
```
Comment (sharkich) [inline homework3/index.js]: What are you returning?

File: homework3/index.js [hunk 3]
```diff
@@ -45,9 +39,10 @@ console.log(updatedPriceData);   // {apples: '23.40', bananas: '48.00', oranges:
 function recursiveOddSumTo(number) {
     if (number === 1){
         return 1;
-    }else if((number%2 == 1)){
-        return number + recursiveOddSumTo(number - 2);
-    } return recursiveOddSumTo(number-1);
+    } if((number%2 == 0)){
```
Comment (sharkich) [inline homework3/index.js]: new `if` - new line

File: homework3/index.js [hunk 4]
```diff
@@ -1,28 +1,22 @@
...
+    const diffTime = Math.abs(firstInputDate - secondInputDate);
+
+    dimensionObject = new Map([
+        ['seconds', 1000],
+        ['minutes', 60000],
+        ['hours', 3600000],
+        ['days', 86400000]
+    ]);
+    if (dimensionObject.has(dimension)){
+        return console.log('Time difference is ' + Math.ceil(diffTime/dimensionObject.get(dimension)) +' '+ dimension);
```
Comment (sharkich) [inline homework3/index.js]: What this function is returning?

**Review by sharkich — CHANGES_REQUESTED** [review]

File: homework3/index.js
Comment (Lubovgroma) [inline homework3/index.js:1]: I have added default values, removed console.log function from return. I assume that func must return always return a value, correct?

File: homework3/index.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 //---------------task 1----------------
-function durationBetweenDates(firstDate, secondDate, dimension) {
+function durationBetweenDates(firstDate = '1999-12-01', secondDate = '2015-05-26', dimension = 'days') {
```
Comment (Lubovgroma) [inline homework3/index.js:2]: added default values

File: homework3/index.js [hunk 6]
```diff
@@ -11,13 +11,13 @@ function durationBetweenDates(firstDate, secondDate, dimension) {
         ['days', 86400000]
     ]);
     if (dimensionObject.has(dimension)){
-        return console.log('Time difference is ' + Math.ceil(diffTime/dimensionObject.get(dimension)) +' '+ dimension);
+        return Math.ceil(diffTime/dimensionObject.get(dimension));
     }
-    return console.log('Wrong dimension');
+    return diffTime;
 }
```
Comment (Lubovgroma) [inline homework3/index.js:18]: removed console.log function from return

**Pull request merged**